### PR TITLE
Music: small rendering fix

### PR DESCRIPTION
**Pull request by breville**
Fixes a React warning by ensuring we have unique keys for lists of elements in the Timeline.

### CODE REVIEW

**Review by thomasoniii — APPROVED** [review]
lgtm 🚀

File: apps/src/music/views/TimelineSimple2Events.tsx [hunk 1]
```diff
@@ -173,6 +173,7 @@ const TimelineSimple2Events: React.FunctionComponent<
       <div id="timeline-events-function-extents">
         {uniqueFunctionExtentsArray.map((functionExtents, index) => (
           <FunctionExtentsSimple2
+            key={index}
```
Comment (levadadenys) [inline apps/src/music/views/TimelineSimple2Events.tsx:175]: Don't have much context on this array, but wanted to ask if the order of this array may change?
Using element's index should be ok if array order stays the same, though if the order might be changed - might potentially cause some bugs on reordering - more info here
https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key

Comment (breville) [reply apps/src/music/views/TimelineSimple2Events.tsx:175]: Very interesting point.  As the Timeline content changes, it is certainly possible that keeping these keys consistent between renders would be counterproductive.  

I've updated the PR with a new approach that ensures we have unique values for each new execution of the code, which is what drives the major changes to the layout of the Timeline.  Am curious if it's sufficient for now, though I'm sure there are cleverer ways we could render (and optimize rendering) of this Timeline in the future.

Comment (breville) [reply apps/src/music/views/TimelineSimple2Events.tsx:175]: We've actually gone back to the simpler approach for now.  @thomasoniii and I talked things through, and we think that it's probably alright to use the index, and it keeps things simpler as we approach launch.  We'll keep an eye out for rendering issues, though.

**Review by sanchitmalhotra126 — APPROVED** [review]

**Pull request merged**